src/main/java/dev/openfeature/sdk/OpenFeatureAPI.java

```diff
@@ -18,9 +20,10 @@ public class OpenFeatureAPI {
     static AutoCloseableReentrantReadWriteLock hooksLock = new AutoCloseableReentrantReadWriteLock();
     static AutoCloseableReentrantReadWriteLock providerLock = new AutoCloseableReentrantReadWriteLock();
     static AutoCloseableReentrantReadWriteLock contextLock = new AutoCloseableReentrantReadWriteLock();
-    private FeatureProvider provider;
     private EvaluationContext evaluationContext;
     private List<Hook> apiHooks;
+    private Map<String, FeatureProvider> providers = new ConcurrentHashMap<>();
```

Comment:
Another alternative here is to make the default provider a non-named provider

```java
private FeatureProvider provider; // this is default provider
private Map<String, FeatureProvider> providers = new ConcurrentHashMap<>(); // these are named providers 
```
This helps to avoid `DEFAULT_PROVIDER_KEY` usage and avoid Map lookups for default usage (which is one less lookup to be performant)